lib/miner-info.js

```diff
@@ -29,6 +30,32 @@ async function getChainHead ({ maxAttempts = 5 } = {}) {
  * @returns {Promise<string>} Miner's PeerId, e.g. `12D3KooWMsPmAA65yHAHgbxgh7CPkEctJHZMeM3rAvoW8CZKxtpG`
  */
 export async function getMinerPeerId (minerId, { maxAttempts = 5 } = {}) {
+  // Make a concurrent request to both sources: RPC node and smart contract
+  const [rpcNodeResult, contractResult] = await Promise.allSettled([
+    getMinerPeerIdFromRpcNode(minerId, { maxAttempts }),
+    getMinerPeerIdFromSmartContract(minerId)
+  ])
+  // Check contract result first
+  if (contractResult.status === 'fulfilled' && contractResult.value) {
+    return contractResult.value
+  }
+
+  // Fall back to rpc-Node result
+  if (rpcNodeResult.status === 'fulfilled' && rpcNodeResult.value) {
+    return rpcNodeResult.value
+  }
+
+  // Handle the case where both failed
+  throw new Error(`Failed to resolve PeerID from any source: ${contractResult.reason}, ${rpcNodeResult.reason}`)
+}
+
+/**
+ * @param {string} minerId A miner actor id, e.g. `f0142637`
+ * @param {object} options
+ * @param {number} [options.maxAttempts]
+ * @returns {Promise<string>} Miner's PeerId, e.g. `12D3KooWMsPmAA65yHAHgbxgh7CPkEctJHZMeM3rAvoW8CZKxtpG`
+ */
+export async function getMinerPeerIdFromRpcNode (minerId, { maxAttempts = 5 }) {
```

Comment:
This name is not correct.
- Both functions are calling RPC API node
- In this function, we are getting the value from Filecoin's MinerInfo state - see https://lotus.filecoin.io/reference/lotus/state/#stateminerinfo

Proposed name:

```suggestion
export async function getMinerPeerIdFromFilecoinMinerInfo (minerId, { maxAttempts = 5 }) {
```

Reply:
Addressed in https://github.com/CheckerNetwork/spark-checker/commit/3a60176172843234a404b43adc8bdb9f82aa494d